client/components/map/NearbyFacilitiesPanel.tsx

```diff
@@ -0,0 +1,165 @@
+import { Box, VStack, Flex, Button, Text } from "@chakra-ui/react";
+import { COLORS } from "theme";
+import { Pages } from "spa-pages/pageEnums";
+import { ChevronLeftIcon } from "@chakra-ui/icons";
+import { Location } from "components/home/UserInput/Location";
+import { HamburgerIcon, Search2Icon, CloseIcon } from "@chakra-ui/icons";
+
+import { IconButton } from "@chakra-ui/react";
+import { Dispatch, SetStateAction, MouseEventHandler } from "react";
+import { RecyclingLocationResults } from "app-context/UserSelectionContext/types";
+import { TStateFacilities } from "app-context/SheetyContext/types";
+import { Image } from "@chakra-ui/react";
+
+interface ButtonProps {
+	onClick: MouseEventHandler<HTMLButtonElement>;
+}
+// Button Design for the filter button
+export const FilterButton = ({ onClick }: ButtonProps) => {
+	return (
+		<IconButton
+			variant="solid"
+			color={COLORS.gray[700]}
+			background={COLORS.gray[100]}
+			aria-label="add line"
+			icon={<HamburgerIcon />}
+			onClick={onClick}
+		/>
+	);
+};
+export const SearchButton = ({ onClick }: ButtonProps) => {
+	return (
+		<IconButton
+			variant="solid"
+			color={COLORS.gray[700]}
+			background={COLORS.gray[100]}
+			aria-label="add line"
+			icon={<Search2Icon />}
+			onClick={onClick}
+		/>
+	);
+};
+export const XButton = ({ onClick }: ButtonProps) => {
+	return (
+		<IconButton
+			variant="solid"
+			color={COLORS.gray[700]}
+			background={COLORS.gray[100]}
+			aria-label="add line"
+			icon={<CloseIcon />}
+			onClick={onClick}
+		/>
+	);
+};
+type NearbyFacilitiesType = {
+	setPage: Dispatch<SetStateAction<Pages>>;
+	handleChangedLocation: () => void;
+	showFilter: () => void;
+	nearbyLocations: RecyclingLocationResults;
+	getMatchingFacility: (facility: { id: number; distance: number; latlng: Array<number> }) => {
+		cardDetails: TStateFacilities;
+	};
+	isMobile: boolean | undefined;
+};
+
+export const NearbyFacilitiesPanel = ({
+	setPage,
+	handleChangedLocation,
+	showFilter,
+	nearbyLocations,
+	getMatchingFacility,
+	isMobile,
+}: NearbyFacilitiesType) => {
+	return (
+		<Box
+			position={"fixed"}
+			height={isMobile ? "calc(92vh - 53px)" : "calc(90vh - 53px)"}
+			width={"100vw"}
+			bg="white"
+			zIndex={1000}
+			bottom={0}
+			overflowY={"scroll"}
+			overflowX={"hidden"}
+		>
+			<VStack p={7}>
+				<Flex flexDir={"row"} w={"100%"} gap={3}>
+					<Button
+						bg={COLORS.Button.primary}
+						color={COLORS.white}
+						leftIcon={<ChevronLeftIcon />}
+						onClick={() => setPage(Pages.HOME)}
+					>
+						Restart!
+					</Button>
+					<Location showText={false} handleBlur={handleChangedLocation} />
+					<FilterButton onClick={showFilter} />
+				</Flex>
+				<Flex flexDir={"column"} maxW={"100vw"} p={5}>
+					{nearbyLocations.results &&

```

Comment:
this whole block looks very deeply nested. is it possible to break it up into components?

Reply:
This component is being archived (it was the component to pull up and show nearby facilities, but design changed)